From 17892e85309db8cf0fd39ca8be24cbc654e11b9b Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
Date: Sat, 9 Aug 2025 14:50:38 +0100
Subject: [PATCH 5/7] Make xpack settings default to FIPS

Set FIPS mode default to true, and set required provider list.
---
 .../org/elasticsearch/xpack/core/XPackSettings.java | 13 +++++--------
 1 file changed, 5 insertions(+), 8 deletions(-)

diff --git a/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackSettings.java b/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackSettings.java
index bfa593afac5..443c306ce74 100644
--- a/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackSettings.java
+++ b/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackSettings.java
@@ -177,13 +177,16 @@ public class XPackSettings {
     /** Setting for enabling or disabling FIPS mode. Defaults to false */
     public static final Setting<Boolean> FIPS_MODE_ENABLED = Setting.boolSetting(
         "xpack.security.fips_mode.enabled",
-        false,
+        true,
         Property.NodeScope
     );
 
+    private static final List<String> DEFAULT_FIPS_REQUIRED_PROVIDERS = List.of("BCFIPS", "BCJSSE", "BCRNG");
+
     /** Optional setting to prevent startup if required providers are not discovered at runtime */
     public static final Setting<List<String>> FIPS_REQUIRED_PROVIDERS = Setting.stringListSetting(
         "xpack.security.fips_mode.required_providers",
+        DEFAULT_FIPS_REQUIRED_PROVIDERS,
         Property.NodeScope
     );
 
@@ -250,13 +253,7 @@ public class XPackSettings {
 
     public static final Setting<String> PASSWORD_HASHING_ALGORITHM = defaultStoredPasswordHashAlgorithmSetting(
         "xpack.security.authc.password_hashing.algorithm",
-        (s) -> {
-            if (XPackSettings.FIPS_MODE_ENABLED.get(s)) {
-                return Hasher.PBKDF2_STRETCH.name();
-            } else {
-                return Hasher.BCRYPT.name();
-            }
-        }
+        (s) -> Hasher.PBKDF2_STRETCH.name()
     );
 
     public static final Setting<String> SERVICE_TOKEN_HASHING_ALGORITHM = defaultStoredPasswordHashAlgorithmSetting(
-- 
2.48.1

